src/Symfony/Component/Security/Guard/AuthenticatorInterface.php

```diff
@@ -94,14 +94,15 @@ public function getUser($credentials, UserProviderInterface $userProvider);
      *
      * The *credentials* are the return value from getCredentials()
      *
-     * @param mixed         $credentials
-     * @param UserInterface $user
+     * @param mixed                     $credentials
+     * @param UserInterface             $user
+     * @param PasswordUpgraderInterface $passwordUpgrader
      *
      * @return bool
      *
      * @throws AuthenticationException
      */
-    public function checkCredentials($credentials, UserInterface $user);
+    public function checkCredentials($credentials, UserInterface $user/*, PasswordUpgraderInterface $passwordUpgrader = null*/);
```

Comment:
What does this mean? To avoid bc break?

Reply:
yes, that's how we avoid BC breaks: new argument in the docblock + inline comment like this one.